searchcore/src/tests/proton/matching/matching_test.cpp

```diff
@@ -528,6 +545,87 @@ TEST("require that re-ranking is performed (multi-threaded)") {
     }
 }
 
+TEST("require that re-ranking is not diverse when not requested to be.") {
+    MyWorld world;
+    world.basicSetup();
+    world.setupSecondPhaseRanking();
+    world.basicResults();
+    SearchRequest::SP request = world.createSimpleRequest("f1", "spread");
+    auto mtf = world.create_mtf(request);
+    auto diversity = mtf->createDiversifier();
+    EXPECT_FALSE(diversity);
+}
+
+using namespace search::fef::indexproperties::matchphase;
+TEST("require that re-ranking is diverse with diversity = 1/1") {
+    MyWorld world;
+    world.basicSetup();
+    world.setupSecondPhaseRanking();
+    world.basicResults();
+    SearchRequest::SP request = world.createSimpleRequest("f1", "spread");
+    auto mtf = world.create_mtf(request);
+    auto & rankProperies = request->propertiesMap.lookupCreate(MapNames::RANK);
+    rankProperies.add(DiversityAttribute::NAME, "a2")
+                 .add(DiversityMinGroups::NAME, "3")
+                 .add(DiversityCutoffStrategy::NAME, "strict");
+    mtf = world.create_mtf(request);
+    auto diversity = mtf->createDiversifier();
+    EXPECT_TRUE(diversity);
```

Comment:
I think activation check should also be put in a separate test

Reply:
Fixed in #6546